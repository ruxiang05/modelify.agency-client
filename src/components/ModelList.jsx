import React from 'react';
import api from '../api';
import ModelCard from './ModelCard';
import { getToken } from '../auth';

class ModelList extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      models: [],
    };
  }

  componentDidMount() {
    const token = getToken();
    api.agents.getModels(token).then((res) => {
      const { models } = res.agent.agentInfo;
      this.setState({ models });
    });
  }

  render() {
    const { models } = this.state;
    return (
      <React.Fragment>
        {models ? (
          <ul>
            {models.map(model => (
              <li key={models.indexOf(model)}>
                <ModelCard {...model} />
              </li>
            ))}
          </ul>
        ) : (
          <div>No models yet</div>
        )}
      </React.Fragment>
    );
  }
}

export default ModelList;
