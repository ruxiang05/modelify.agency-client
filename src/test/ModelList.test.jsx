import React from 'react';
import { shallow } from 'enzyme';
import ModelList from '../components/ModelList';

jest.mock('../api.js');

describe('ModelList', () => {
  it('should render ModelList', async () => {
    const wrapper = shallow(<ModelList />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a list of models', async () => {
    const wrapper = shallow(<ModelList />);
    await wrapper.update();
    expect(wrapper.state().models.length).toBe(1);
    expect(wrapper.find('li').length).toBe(1);
  });
});
