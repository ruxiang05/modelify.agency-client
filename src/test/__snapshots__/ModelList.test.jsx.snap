// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModelList should render ModelList 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModelList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="no-data"
      >
        <ForwardRef />
        <p>
          You do not have any models yet
        </p>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef />,
          <p>
            You do not have any models yet
          </p>,
        ],
        "className": "no-data",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You do not have any models yet",
          },
          "ref": null,
          "rendered": "You do not have any models yet",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="no-data"
        >
          <ForwardRef />
          <p>
            You do not have any models yet
          </p>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ForwardRef />,
            <p>
              You do not have any models yet
            </p>,
          ],
          "className": "no-data",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "You do not have any models yet",
            },
            "ref": null,
            "rendered": "You do not have any models yet",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
